### PR TITLE
Add pools(cspc) listing feature to the present CLI.

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
**What will this PR do?**
- This will add the placeholder for addition of pools related features.
- This will add a new command ` kubectl openebs cStor pool list` for listing the CSPIs.

**The command and it's output**
```
k8s@master-sh2:~openebsctl$ kubectl openebs cStor pool list
Name                  Namespace  HostName     Free    Capacity   ReadOnly  ProvisionedReplicas  HealthyReplicas  Status  Age
----                  ---------  --------     ----    --------   --------  -------------------  ---------------  ------  ---
cstor-storage-1-2dr2  openebs    worker-sh23  48200M  48200226k  false     0                    0                ONLINE  3h11m
cstor-storage-1-xhlh  openebs    worker-sh1   48200M  48200221k  false     0                    0                ONLINE  2h42m
cstor-storage-fq44    openebs    worker-sh23  48200M  48205630k  false     1                    1                ONLINE  3d22h


```

### CODE REVIEW

File: kubectl-openebs/cli/command/cstor/pool/pool_list.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	
+	Usage: kubectl openebs cStor pool list [options]
+	`
+	namespace string
+)
+
+// NewCmdPoolsList displays status of OpenEBS Pool(s)
+func NewCmdPoolsList() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
```
[the remark this answers is not in the document]
Comment (vharsh) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:42]: In upcoming PRs we plan to drop `cstor` from the command as discussed today.

File: client/k8s.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2020 The OpenEBS Authors
+Copyright 2021 The OpenEBS Authors
```
Comment (kmova) [inline client/k8s.go:2]: make this 2020-2021

Comment (Abhinandan-Purkait) [reply client/k8s.go]: - Changed to 2020-2021.

File: kubectl-openebs/cli/command/cstor/pool/pool_list.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	return cmd
+}
+
+//RunPoolsList fetchs & lists the pools
+func RunPoolsList(cmd *cobra.Command) error {
+
+	client, err := client.NewK8sClient(namespace)
+	util.CheckErr(err, util.Fatal)
+
+	cpools, err := client.GetcStorPools()
```
Comment (mittachaitu) [inline kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: Command says list about cStor pools but we are listing CStorPoolClusters... Is that expected?

Comment (Abhinandan-Purkait) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: - cStor pools are denoted by CSPCs as far as I know. If we are talking about listing both CSPCs and SPCs then we are listing only CSPCs as moving towards the latest versions of OpenEBS SPCs would be deprecated.

Comment (vharsh) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: @mittachaitu Did you mean pools as in pool-instances and not pool clusters?

Comment (mittachaitu) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: Yes, I should have specified the term `cstorPoolInstance` in the earlier comments.

Actually, In cStor there are more confusing terms(this will get clear off when one started using/working on that). What I usually think(Not sure whether it is valid or not):
cStor pools :-  Actual underlying pools on disk (User no need to worry about this).
Pool Instances/cStor Pool Instances:- Abstraction/representation of underlying disk pools(Represents various stats of underlying pool).
cStorpoolcluster/cStor cluster:- Represents a group of pool instances.

Basically, CLI sub-commands will depends on what we have to project.

Comment (Abhinandan-Purkait) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: @mittachaitu In the subsequent PRs, would were planning to add a describe feature which takes a `cspc` name and displays all sorts of information, related to its `instances`. 
Would that be more meaningful or shall we list the instances in this `list` command and for describe we would take the name of the instance and display details related to it. 
These are two scenarios we can have. Please let us know which suits the purpose more.

Comment (mittachaitu) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: Here, when the describe of CSPC command is triggered and if it displays info about instances makes sense(since motive is to get enough information by running single command). 

IMO we can make it more simple and clear in the following way(**_Not sure how far below pattern is valid_**):

- Displaying all the CSPCs and their information present in the cluster
```sh
kubectl openebs get cStor poolcluster
Name           Namespace  HealthyInstances  ProvisionedInstances  DesiredInstances  Age
----           ---------  ----------------  --------------------  ----------------  ---
cstor-storage-1  openebs    1                 1                     1                 3d27m
cstor-storage-2  openebs    1                 1                     1                 3d27m
cstor-storage-3  openebs    1                 1                     1                 3d27m
```

- Displaying the CSPIs information when a user specifies CSPC name along with the command (I not sure whether it is valid or not). 
```sh
kubectl openebs get cStor poolcluster <cstor-storage-1>
Name           Namespace  UsedSpace  AvailableSpace TotalSpace TotalReplicas HealthyReplicas  Age
----           ---------  ----------------  --------------------  ----------------  ---
...
...
...
```
- Above pattern will avoid introducing other sub-commands like poolinstance.

Comment (Abhinandan-Purkait) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: - Displaying all the CSPCs and their information present in the cluster
   For the above we use the verb `list` in place of `get`.

- Displaying the CSPIs information when a user specifies CSPC name along with the command 
  For the above we would use the verb `describe` in place of `get`. 

- We will change the term from `pool` to `poolcluster`

@mittachaitu If these seems fine we can finalize on this.

Comment (mittachaitu) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: Make sense :+1:

Comment (Abhinandan-Purkait) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: Thank you.

Comment (kmova) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: If we are listing `CSPI` then use `pool`, if using `CSPC` then using PoolCluster .. makes sense.. but let us avoid using Pool Cluster -- as it not a commonly understood term.

Comment (Abhinandan-Purkait) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: - We were listing `CSPCs` initially, as `CSPCs` are a group of instances, which upon description would have brought out the underlying `CSPIs`, but seems like `CSPIs` would be the requirement as a list. I will keep the command same i.e. `kubectl openebs cStor pool list` and change the output from `CSPCs` to `CSPIs`.

Comment (Abhinandan-Purkait) [reply kubectl-openebs/cli/command/cstor/pool/pool_list.go:64]: - I have updated the PR description, with the latest output of this command.

**Review by kmova — APPROVED** [review]

**Comment by vharsh** [comment]
Please fix the build errors viz
- golint errors (GoLand does a really bad job in running golint, gofmt, etc or maybe I don't know how to configure it correctly yet)
- the copyright header is missing in the test files, so the `Verify license` stage is failing
- run `go mod vendor` and check-in the go.mod & go.sum files once again to fix below errors in CI as steps `run GoLint`, here're the some errors from the build

```logs
go/build: go list github.com/openebs/openebsctl/kubectl-openebs: exit status 1
go: inconsistent vendoring in /home/runner/work/openebsctl/openebsctl:
	golang.org/x/lint@v0.0.0-20201208152925-83fdc39ff7b5: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/tools@v0.1.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/net@v0.0.0-20200625001655-4c5254603344: is marked as explicit in vendor/modules.txt, but not explicitly required in go.mod
```

**Review by shubham14bajpai — APPROVED** [review]
LGTM
